drivers/hmis/app/models/hmis/hud/client.rb

```diff
@@ -54,6 +54,10 @@ class Hmis::Hud::Client < Hmis::Hud::Base
   has_many :hmis_services, through: :enrollments # All services (HUD and Custom)
   has_many :services, through: :enrollments # HUD Services only
   has_many :custom_services, through: :enrollments # Custom Services only
+  has_many :enrollments_with_chronic_homelessness,
+           -> { where(chronically_homeless_at_entry: true) },
+           through: :enrollments,
+           source: :ch_enrollment
```

Comment:
Is there a description somewhere about what `chronically_homeless_at_entry: true` means?

From looking at the code, I think it means "if the enrollment is exited, client was deemed chronically homeless on the day of exit. If the enrollment is open, client is deemed chronically homeless today." I'm interpreting that from [the update_existing code](https://github.com/greenriver/hmis-warehouse/blob/46e2ed52a4ece023b13be213bd2dcec65b9f81d9/app/models/grda_warehouse/ch_enrollment.rb#L66-L71) and how it chooses date. But the name of the field itself is `chronically_homeless_at_entry`.. I don't see entry date being used in the actual calculation.

Reply:
Maybe we can give the relation a more descriptive name, like `enrollments_with_chronic_today_or_at_exit`  🫣 or, filter to open enrollments only and call this `open_enrollments_hud_chronic_at_point_in_time` or `open_enrollments_with_hud_chronic_today` etc

Reply:
@gigxz this calculation should match up with **CH at Project Start* from the glossary for any project where the project type is not 0, 1, 4, or 8.  For those projects (homeless projects) I believe it should align with **CH at a Point In Time**.  The column name is definitely confusing.

Reply:
Can you point me to the code where that braching happens? I'm seeing that the function `chronically_homeless_at_start_steps` receives the same `date` to work from, regardless of project type. I see that the project type affects some internal calculations, like how `total_months_homeless` is calculated, but ultimately we are still calculating whether they were considered chronic at the point-in-time (which is [either Today or Exit Date](https://github.com/greenriver/hmis-warehouse/blob/46e2ed52a4ece023b13be213bd2dcec65b9f81d9/app/models/grda_warehouse/ch_enrollment.rb#L66)).

Reply:
maybe I'm missing something but these two comments appear to be conflicting, and I think the first one may be inaccurate..
```
    # Accept an optional date which will be used for extending the homeless
    # range if the project is a homeless project
```
...
```
    # Optionally accepts a date to use for "CH at a point-in-time" calculation.
```

Reply:
fwiw I'll probably just remove the `enrollments_with_chronic_homelessness` scope and handle it in the hud_chronic field resolver instead

Reply:
Ok. We should still wait to hear from @eanders because if it _does_ work in the way he described (CH at Entry for certain project types, CH at P-I-T for others) then this implementation does NOT match up with the tooltip displayed in the frontend, which says that it's displaying chronicity statues for point-in-time of today.

Reply:
So, the calculation is:
1. Limited to the associated enrollment
2. aware of the current date (but can be fed a different date)
3. cached/calculated daily - meaning the data that is stored will represent **CH at Project Start** for non-homeless projects (which is actually the same thing as **CH at a Point In Time** since you don't acquire homeless time being in a non-homeless project.  BUT, in homeless projects, you do acquire days for every day you are there.  So joining `ChEnrollment` is the equivalent of **CH at a Point In Time** for each enrollment.  

Does that make sense?

Reply:
Thanks for explaining, I'll capture the above in code comments. Maybe we can discuss this at dev stand up.

Reply:
Can we huddle on this?

I think I get that what you're saying matches up with the reporting glossary (quote below), but I don't fully understand:
1. how this is reflected in the code
2. how we should be communicating this new value to the user. It no longer feels accurate to present this as answering "is this client considered chronic by HUD today?".. it's more like, "were they chronic at entry at any of their open non-homeless enrollments, or are they chronic today based on homeless enrollments?"


> Households in street outreach, shelters, and safe havens ([project type] = 0, 1, 4, or 8) may also have a “point-intime” CH status designed to include time spent from [project start date] up to a [point-in-time date] as time spent experiencing homelessness. Clients in projects other than street outreach, shelters, and safe havens may only have a CH status at project start and cannot age into CH by virtue of being in any of those projects. Reporting on point intime CH for these other project types simply uses the client’s CH status at project start.



The original implementation of this field used `hud_chronic?`, which matched [the implementation of](https://github.com/greenriver/hmis-warehouse/blob/a96126ec80ec9f3e1069ce490afc74580cd0407e/app/views/clients/chronic/_chronic_calculator.haml#L20) the Chronic tab on the Warehouse Client dashboard. It was nice that those two matched because you could go over and see the explanation (and validate that they match), provided that you have the same level of access. The Chronic tab presents the data in much the same way we do in HMIS-- answering the question "is this Client considered HUD Chronic on this Date?". It's still not clear to me how the ChEnrollment answers this -- and why we don't use the same implementation on the Warehouse Chronic tab, since it appears to be presenting the same type of information.

<img width="1485" alt="Screenshot 2024-11-25 at 12 34 39 PM" src="https://github.com/user-attachments/assets/8b66e9f1-069c-4cee-b2f6-eede52d3cd57">

Reply:
maybe we just need to further limit the enrollment scope to `ES (1), SH (8) or SO (4) and not currently housed (other residential types)`

Reply:
discussion:

1) return false immediately if they have an open residential project with a move in date that has occurred, which indicates that they are housed
2) return false immediately if they have no open enrollments at homeless project types
3) limit the scope of enrollments to open enrollments, join to ChEnrollment and look for the `chronically_homeless_at_entry`